### PR TITLE
[UCDC] Fix profile 500 error. Add new field.

### PR DESCRIPTION
**Pull request by arsentur**
**Description:**
- Fix 500 error on the profile page
- Add a new diploma field that relates to the Position and SubPosition fields.
- Add phone and date of birth validation.

**Youtrack:**
[UCDC-177](https://youtrack.raccoongang.com/issue/UCDC-177) - 500 on the profile page
[UCDC-202](https://youtrack.raccoongang.com/issue/UCDC-202) - New field
[UCDC-208](https://youtrack.raccoongang.com/issue/UCDC-208) - Problem with getting certs

### CODE REVIEW

File: common/djangoapps/student/models.py [hunk 1]
```diff
@@ -434,14 +436,15 @@ class Meta(object):
...
-    date_of_birth = models.DateField(blank=True, null=True, db_index=True)
+    date_of_birth = models.DateField(blank=True, null=True, db_index=True, validators=[validate_date_of_birth])
 
     REGION_CHOICES = [(region[0], _(region[1])) for region in getattr(settings, 'REGIONS', [])]
     region = models.CharField(max_length=255, null=True, blank=True, choices=REGION_CHOICES)
     position = models.ForeignKey(Position, null=True, blank=True, on_delete=models.CASCADE)
     other_position = models.CharField(max_length=255, null=True, blank=True)
 
     specialization = models.ManyToManyField(Specialization, blank=True)
+    diploma_number = models.CharField(max_length=11, blank=True, null=True)
```
Comment (NikolayBorovenskiy) [inline common/djangoapps/student/models.py]: Are we sure that 11 is the optimal limit?

Comment (NikolayBorovenskiy) [reply common/djangoapps/student/models.py]: And here we can get non-numeric symbols, right?

File: common/djangoapps/student/models.py [hunk 2]
```diff
@@ -434,14 +436,15 @@ class Meta(object):
     phone = models.CharField(max_length=20, null=True, blank=True)
     second_name = models.CharField(max_length=100, null=True, blank=True)
     additional_email = models.EmailField(max_length=255, null=True, blank=True)
-    date_of_birth = models.DateField(blank=True, null=True, db_index=True)
+    date_of_birth = models.DateField(blank=True, null=True, db_index=True, validators=[validate_date_of_birth])
```
Comment (NikolayBorovenskiy) [inline common/djangoapps/student/models.py]: I see that you add the validator directly on the model. Do not we have form to validate there?

File: common/djangoapps/student/models.py (hunk 1 above)
Comment (arsentur) [reply common/djangoapps/student/models.py]: Yes, you're right. The diploma number can be in the following formats: 
[Two options: АА000000 (старые дипломы) or АА00000000 (новые дипломы)]

File: common/djangoapps/student/models.py (hunk 2 above)
Comment (arsentur) [reply common/djangoapps/student/models.py]: Yes, we can. I thought that will be a good idea.

Comment (arsentur) [reply common/djangoapps/student/models.py]: Done

**Pull request merged**